### ISSUE TITLE
Balance between taxes vs food selling.

### ISSUE THREAD

**Issue by Mizo5119** [Sh0wdown/UnofficialCrusaderPatch#416]
I was looking up some old info collected and I found this blog post:

http://stronghold.heavengames.com/cgi-bin/forum/display.cgi?action=ct&f=14,3439,,30

These are actually valid points that are brought up regarding taxes vs food. It is simply more cost efficient to sell food and use beer to cover up the popularity loss than to rely on taxes to generate income. 

This seems very "odd" for a sim castle to encourage users to sell food. Can the tax rates be buffed so that they arent an inferior source of income?

**Comment by Rakso69** [comment]
Actually being honest I like that atleast these first two strongholds are different, because on every other Stronghold game(except for SH2 when mass bread selling I guess) you're relying solely on taxes and you don't really need anything more unless for like a bonus to "make you purchase less, to safe gold from taxes in overall", because selling anything sucks as hell.
Nevertheless saying that selling is better than taxes, doesn't mean that taxes are bad, they should be used aswell.
Also the whole post is invalid, because taxes are different depended of either if the AI has the gold advantage, or the human player: https://github.com/Sh0wdown/UnofficialCrusaderPatch/issues/382

**Comment by GRhin** [comment]
First off, @Rakso69 the different tax benefits does not make this post invalid, merely situational. According to the math linked, taxes are worse, and that only changes when you give the human the advantage (unless using this math for AI), and how often are players giving themselves the advantage and also streamlining gold to the level of selling food?
That being said, I think the math linked is actually wrong.
@Mizo5119 I think you need to do more research on this, in that post artofmath said that he didnt take into account selling ale. In fact, theres a few things he didnt account for, the biggest oversight was ignoring ale with food, so i will illustrate some scenarios using his math (im not testing myself, so dont quote me on this). Asuuming the same number of workers producing both food and ale in all situations, and providing for 90 workers:
Scenario 1: No rations, no tax, max ale, sell food.
In this case, you are saving 14.4 food per month vs double rations (0.16 food per month per worker, according to artofmath), which sells for 57.6 gold
Scenario 2: Normal rations, -8 tax, max ale
Saves 7.3 food per month compared to extra rations (0.08 food per month per worker), which sells for 29.2 gold. Plus 90 gold from taxes (1gpm according to the thread rakso linked). this is a bonus 61.6 gold per month compared to scenario 1.
Scenario 3: Double Rations, -16 tax, max ale
I dont know the exact rate of -16 tax, but im assuming around 1.5gpm (it is 1gpm at -8, and 2gpm at -24 according to the other thread) which gives us 135 gold per month, which is an extra 15 gold per month compared to scenario 2.

So for streamlining gold production, selling food isnt actually better. That being said, there are lots of benefits of going with beer:
1. It uses less population to run, allowing you to use more workers on more valuable tasks like weapon creation or getting raw material (stone/iron/wood), 
2. It takes less oasis space, so if you are short on that, then prioritise hops over food,
3. It uses less production space than bread production, making it easier to enclose all your production buildings in a castle.
4. It costs less to purchase enough hops or beer to maintain coverage than it does to purchase food, so if you have no access to oasis, go for beer instead.

Put simply, if you have to pick between beer or food, pick beer, but if you are producing food as well, tax from it instead.

Just to point out exactly why my results are different from the post you linked; artofmath compared maxfood with no ale, against no food with ale coverage - he added a variable for one situation, but not the other. I assumed ale coverage in all scenarios.

**Comment by Krarilotus** [comment]
Ok, so i ve tested the following:
250 population
-> normal tax (no human advantage)
-> 50g / taxtier, 150g for first tax tier (+1 -> -2) or 100g for first positiv tier (+3 to +1)
-> makes the taxes as follows:
popularity | tax
+7 | -200
+5 | -150
+3 | -100
+1 | +0
-2 | +150
-4 | +200
-6 | +250
-8 | +300
-12 | +350
-16 | +400
-20 | +450
-24 | +500
-> normal rations food usage per month: 250/6 = 41,666
-> inefficient but normal inn placement: 0.16666 beer  per inn / month
-> no fear factor applied

prices of the different goods:
**buying:**
buy wood = 20g/5wood -> 4g / wood
buy wheat = 115g/5wheat -> 23g/ wheat
buy hop = 75g/5hop -> 15g/ hop
buy ale = 100g/5ale -> 20g/ale
buy flour = 160g/5flour -> 32g/flour

**selling:**
wheat = 40g/5wheat -> 8g/wheat
hop = 40g/5hop -> 8g/hop
flour = 50g/5flour -> 10g/flour
beer = 50g/5beer -> 10g/beer
bread = 20g/5bread -> 4g/bread
crossbows = 150g/5crossbows -> 30g/crossbow

**production ratios:**
2wood -> 1crossbow
1hop -> 1beer
1wheat -> 1flour
1flour -> 8bread

**food production test:**
10 years = 120 months (equals 5.000 bread consumed)
9 wheat farms, 3 mills, 30 bakerys (effective placement)
building costs: 495 wood -> buying that means 1.980 gold
population cost: 48
-> keep in mind, that this needs usually more pop, since fireworkers are needed in case of fire!
tiles used: 
- 9* 9x9 farmland = 9* 81 = 729 farmland tiles
- 3* 3x3 + 30*4x4 + walking space = 3*9 + 30*16 + walking space = 507 castle tiles + walking space

production roughly 5.000 bread -> sells for 20.000 gold


**Ale production test:**
10 years = 120 months (equals 20 ale per inn)
4 hop farms, 7 brewerys, 9 inns (+8 popularity)
building costs: 180wood + 900 gold -> buying that means 2.140 gold)
population costs: 20
-> place the inns further away to reduce their usage of ale -> makes it more efficient.
tiles used:
- 4* 81 farmland = 324 farmland tiles
- 7*4x4 + 9*5x5 + walking space= 7*16 + 9*25 + walking space= 337 castle tiles + walking space

-> turning off the inns, leaves 11 production workers for ale:
production of 110 hops and 267 beer (180 would have been consumed)
total value sold: 110*8 + 267*10 = 3.550 gold
inn usage beer value sold: 180*10 = 1.800 gold
scale to 48 pop:
total value 42pop: 3.550/11 * 48 = 15.491 gold


**crossbow production test:**
10 years = 120 months
30 fletchers (around the same space needed as for the 30 bakerys -> test if its better than food production)
building costs: 600 wood + 3.000 gold -> buying that means 5.400 gold
population costs: 30
-> use space near stockpile and armory for efficiency
tiles used: 30*16 + walking space = 480 tiles 
-> buying the wood for one cb costs 8g, selling 1 cbgives 30 gold -> 1cb gives income of 22 gold
-> crossbow production needs no farmland!!!!
production: 690 crossbows in 10 years -> 22*690 = 15.280 gold
scale to 48pop:
total 48 income: 15.280 / 30 * 48 = 24.448 gold (for 768 castle tiles tho)



**Scenarios:**
build a cathedral = 13*13 tiles = 169 tiles for +2 popularity (1pop need, 1k gold investment)

**(1)** -2 tax, no rations, ale production + inns enabled, food production, big castle
building costs: around 5.120 gold or 675 wood + 1.900 gold
-> 150g * 120 = 18.000 tax income
-> 20.000 gold bread selling income
-> 1.750 gold excess ale prod income
free pop: 250 - 48 - 20 - 1 = 181
investment costs gained back ~ after 16 months

**(2)** -8 tax, normal rations, ale production+ inns enabled, food production, medium to big castle, no cathedral
building costs: around 4.120 gold or 675 wood + 900 gold
-> 300*120 = 36.000 tax income
-> 1.750 gold excess ale prod income
free pop: 181
investment costs gained back ~ after 13 months

**(3)** -2 tax, no rations, ale prod + inns enabled, no farmland, medium castle, crossbowproduction:
building costs: around 8.540 gold or 780 wood + 4.900 gold
buy the hop! 180 * 15 = addiditional 2.700 investment costs -> total investment each year 270 gold extra
-> 150*120 tax = 18.000 tax
-> 15.280 gold from crossbowprod
free pop: 250 - 30 - 9 - 4 - 1 = 206
investment costs gained back ~ after 34 months

**(4)** -2 tax, no rations, ale prod + inns enabled, no farmland, big castle, crossbowproduction:
building costs: around 11.260 gold or 1.140 wood + 6.700 gold
buy the hop! 180 * 15 = addiditional 2.700 investment costs -> total investment each year 270 gold extra
-> 150*120 tax = 18.000 tax
-> 24.448 gold from crossbowprod
free pop: 250 - 48 - 14 = 188
investment costs gained back ~ after 34 months

feel free to explore more szenarios. I hope this brings some light into all of this @Rakso69 and @GRhin, as you might want to design optimal AI's :D
But actually, this only works, if you use the 250 pop, and dont stay at some lower population. 250 is a good number, but you can make also all these calculations with less pop and or different taxes (times 1.5 for big advantage, times 1.25 for little advantage for the taxes)
I think the taxes are pretty bunkers, but in the end, selling the bread might be better. Its a close call i woud say. It also depends on the farmland that you have. so you might want to calculate a scenario with everything, food prod, ale prod, and crossbow prod, but keep in mind, that your castle might be too big for most maps. And also keep in mind, you can still put the rest of your population later into quarries, ironmines or weapon production like more crossbows, after the main castle is finished. Since the investment for crossbowprodcution is very high compared to food, i would only recommend that if you dont have enough space.

**Comment by Lolasik011** [comment]
A lot of incomprehensible information : D 
@Rakso69 @Krarilotus I will wait from you new AI scripts...

**Comment by GRhin** [comment]
OK, so I didnt fully understand Krarilotus's post, but used his data he put at the top of his post, and fixed a flaw in my math, to get the accurate value of taxes versus selling food. Based off of 90 population, this does not take into account the opportunity cost of making this food/beer (i.e. if you had workers making crossbows instead of food), that is a slightly different discussion. - Skip to the bottom if you just want the results.

**Scenario 1: No rations, no tax, max ale, sell food.**
In this case, you are saving 30 food per month vs double rations (1/3 food per month per worker, according to krarilotus), which sells for 120 gold
**Scenario 2: Half rations, -4 tax, max ale**
Saves 22.5 food per month compared to double rations (3/12 food per month per worker saved), which sells for 90 gold. Plus 72 gold from taxes (0.8gpm according to the stats krarilotus gave). 
=162 gold per month
**Scenario 3: Normal rations, -8 tax, max ale**
Saves 15 food per month compared to double rations (1/6 food per month per worker saved), which sells for 60 gold. Plus 108 gold from taxes (1.2gpm according to the stats krarilotus gave). 
=168 gold per month
**Scenario 4: Extra rations, -16 tax, max ale**
Saves 7.5 food per month compared to double rations (1/12 food per month per worker saved), which sells for 30 gold. Plus 126 gold from taxes (1.4gpm according to the stats krarilotus gave). 
=156 gold per month
**Scenario 5: Double Rations, -16 tax, max ale**
No food sold (any extra food to this would be sold in all scenarios so is not included in maths)
Tax is 144 gold (1.6gpm according to Krariloutus stats
=144 gold per month
 which is 24 gold less than scenario 2, 

The only flaw I see,is that in scenario 1 you have +1 popularity which is not being taxed, so you might want to find a way to get another +1, or you can tax -2 for 1 month and no tax  for 2 months to keep your popularity at a roughly consistent level. Which is (0.6gpm / 3 months = 0.2gpm) an extra 18 gold. But that still does not bring its value up to match Scenario's 2 or 3.


**TLDR;**
no rations = 120 gold per month (or 138 if you taxed 1 month in 3 to take advantage of the no tax popularity bonus)
half rations = 162 gold per month
**normal rations = 168 gold per month – best!**
extra rations = 156 gold per month
double rations = 144 gold per month

**HOWEVER** (fine print and disclaimers below)
#####  thought does need to go into where else your popularity is coming from. if you have lots of faith, for example (i think max is +7 popularity), then the difference between no rations and normal rations is actually closer to the difference between the example above's normal and doubl rations, due to the higher tax levels not being efficient.So if you are getting positive popularity from another source, then selling is better, but if you are getting negative popularity from another source (e.g. bad things or crowding) then increasing rations is better. So again, this is situational. The above numbers are good for a default, just keep this point in mind when you start looking at faith and fearfactor, or even crowding, as being a part of your castle..
